src/components/transactions/SignTxButton/index.tsx

```diff
@@ -11,6 +11,23 @@ import IconButton from '@mui/material/IconButton'
 import CheckIcon from '@mui/icons-material/Check'
 import Track from '@/components/common/Track'
 import { TX_LIST_EVENTS } from '@/services/analytics/events/txList'
+import { txSubscribe, TxEvent } from '@/services/tx/txEvents'
+
+const useIsSignatureProposalPending = (txSummary: TransactionSummary) => {
+  const [isSignatureProposalPending, setIsSignatureProposalPending] = useState<boolean>(false)
+
+  // After a signature proposal, we use the following as a loading flag
+  // because the confirmations array won't update until next poll
```

Comment:
Wait, so the next poll does have the updated list?
Then re-triggering `useLoadTxQueue` alone should fix it?

Reply:
Triggering a poll does not return an updated confirmations list. It takes the backend a few seconds before it is updated.

Reply:
Any suggestion on how I can improve the comment?

Reply:
```
There's a lag between a signature is successfully submitted to the backend, and the tx is updated in the queue. So we need to have a local pending state until the signatures list updates in the queue.
```